### ISSUE TITLE
Come up with better code-sharing across starters

### ISSUE THREAD

**Issue by SethDavenport** [rangle/rangle-starter#32]

**Comment by SethDavenport** [comment]
Right now we have a lot of duplicate configs between starters.  It would be nice to share some of it - candidates include:
- lint rules
- circle.yml
- Procfile
- package.json run scripts
- server.js
- webpack.conf
- collections of dependencies:
  - typescript stuff
  - babel stuff
  - redux stuff
  - react stuff
  - Etc.

If we harmonize the apps we can maybe even centralize the store and reducer logic.

One option would be a generator, but that adds complexity.  Another option would be to beef up the repocopy script from rangle-starter to swap in template configs depending on techstack.

I'm open to suggestions on this one.

**Comment by SethDavenport** [comment]
It would also be nice to make the rangle-starter CI run itself and `npm start` on all four repos to make sure they're functional.

**Comment by SethDavenport** [comment]
One interesting approach might be have the repo copy script ask for a bunch of choices instead of a single tech stack argument - something like:

rangle-starter-2

> repo-url?
> GH username?
> UI ecosystem? [react | angular | angular2]
> transpiler? [babel | react]

... and it would cook up a starter from bits for you.  Remains to be seen if this is less effort or not.

**Comment by kosz** [comment]
are we still not interested in using Slush for some of this ?

**Comment by kosz** [comment]
so @SethDavenport should we try to do anything on this ? 
- First off I'd like to suggest that we keep official rangle repos for our linting, circle files and other somewhat static assets. These files can then be updated into the project, diffed and merged as needed. 
- Secondly, any dynamic/templated code that we may want to generate, could be generated using lodash templates, however once we go down that road we may have to deal with more maintenance that we may want to deal with.

**Comment by SethDavenport** [comment]
This is a low priority right now.

**Comment by JaKXz** [comment]
I just came across: https://github.com/kentcdodds/kcd-common-tools, which I think is a great idea - just have _this_ npm package (or some `rangle-starter-shared` package) have all the files we want shared, and then add it as a dependency and symlink to the given files. 

Thoughts @SethDavenport @kosz? I would make a PR instead of a comment but I really should get back to studying for midterms 😆

**Comment by SethDavenport** [comment]
yeah it's a decent idea, just haven't had time to think through all the permutations.
